lib/actions/local/common.js

```diff
@@ -0,0 +1,99 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as dockerUtils from '../../utils/docker';
+import { exitWithExpectedError } from '../../utils/patterns';
+import { getChalk } from '../../utils/lazy';
+
+export const dockerPort = 2375;
+export const dockerTimeout = 2000;
+
+export const filterOutSupervisorContainer = function(container) {
+	for (let name of container.Names) {
+		if (
+			name.includes('resin_supervisor') ||
+			name.includes('balena_supervisor')
+		) {
+			return false;
+		}
+	}
+	return true;
+};
+
+export const selectContainerFromDevice = Promise.method(function(
+	deviceIp,
+	filterSupervisor,
+) {
+	if (filterSupervisor == null) {
+		filterSupervisor = false;
+	}
+	const form = require('resin-cli-form');
+	const docker = dockerUtils.createClient({
+		host: deviceIp,
+		port: dockerPort,
+		timeout: dockerTimeout,
+	});
+
+	// List all containers, including those not running
+	return docker.listContainers({ all: true }).then(function(containers) {
```

Comment:
Switched to using the native dockerode promise functions because typescript was unaware of our manual promisification and it makes sense to use the native versions now they're available (they still return bluebird promises because docker-toolbelt passes in `{ Promise: Bluebird }`)